lib/superintendent/request/validator.rb

```diff
@@ -97,13 +97,21 @@ def serialize_errors(request_id, form_errors)
     # Determine the requested resource based on the requested endpoint
     def requested_resource(request_path)
       parts = request_path.split('/')
-      # if the last part is an ID, return the part before it, the resource.
-      if (parts[-1]=~/(\d+|[A-Z]{2}[a-zA-Z0-9]{32})/)
-        resource = parts[-2]
+      raw_resource(parts).singularize.classify
+    end
+
+    ID = /(\d+|[A-Z]{2}[a-zA-Z0-9]{32})/
+    RELATION = /^relationships$/
+    def raw_resource(parts)
+      if (parts[-1]=~ ID)
+        # resource/id return resource
+        parts[-2]
+      elsif (parts.size > 3 && parts[-3] =~ ID && parts[-2]=~ RELATION)
+        # matches resource/id/relationships/relationship_resource
+        [parts[-4], parts[-2], parts[-1]].map(&:classify).join('_')
```

Comment:
Feels odd to classify here and do singularize + classify on the return of the method. I'd kindof prefer doing all that in one place or the other.